bin/pyspark

```diff
@@ -42,11 +42,10 @@ if [[ -z "$PYSPARK_DRIVER_PYTHON" ]]; then
   PYSPARK_DRIVER_PYTHON="${PYSPARK_PYTHON:-"python"}"
 fi
 
-WORKS_WITH_IPYTHON=$(python -c 'import sys; print(sys.version_info >= (2, 7, 0))')
-
 # Determine the Python executable to use for the executors:
 if [[ -z "$PYSPARK_PYTHON" ]]; then
-  if [[ $PYSPARK_DRIVER_PYTHON == *ipython* && ! $WORKS_WITH_IPYTHON ]]; then
+  WORKS_WITH_IPYTHON=$($PYSPARK_DRIVER_PYTHON -c 'import sys; print(sys.version_info >= (2, 7, 0))')
```

Comment:
This decides the executor's Python to use. `PYSPARK_DRIVER_PYTHON` shouldn't matter. Also, Spark itself already requires Python 2.7 since we dropped 2.6. So we can just remove this check.

We can also check if both `PYSPARK_DRIVER_PYTHON` and `PYSPARK_PYTHON` are higher than Python 2.7.

Reply:
I found this part indeed very confusing. 
I'll change it back to `WORKS_WITH_IPYTHON=$(python -c 'import sys; print(sys.version_info >= (2, 7, 0))')`
It's already an improvement that this part is inside `if [[ -z "$PYSPARK_PYTHON" ]];`

But indeed, if spark already requires at least py2.7, all this version checking could be reconsidered in the first place

Reply:
I agree that it's confusing ..